server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -106,6 +107,49 @@ public static boolean isSchedulable(JobCurator jobCurator, JobStatus status) {
         return running == 0;  // We can start the job if there are 0 like it running
     }
 
+    private void parseHypervisorList(HypervisorList hypervisors,
+                                     Set<String> hosts,
+                                     Set<String> guests,
+                                     Map<String, Consumer> incomingHosts) {
+        int emptyGuestIdCount = 0;
+        int emptyHypervisorIdCount = 0;
+        ListIterator<Consumer> hypervisorsIterator = hypervisors.getHypervisors().listIterator();
+        while (hypervisorsIterator.hasNext()) {
+            Consumer hypervisor = hypervisorsIterator.next();
+            if (hypervisor.getHypervisorId() != null &&
+                hypervisor.getHypervisorId().getHypervisorId() != null) {
+                if (hypervisor.getHypervisorId().getHypervisorId().equals("")) {
+                    // Ignore hypervisors that have a hypervisor ID of ""
+                    hypervisorsIterator.remove();
+                    emptyHypervisorIdCount++;
+                    continue;
+                }
+                incomingHosts.put(hypervisor.getHypervisorId().getHypervisorId(), hypervisor);
+                hosts.add(hypervisor.getHypervisorId().getHypervisorId());
+                if (hypervisor.getGuestIds() != null && !hypervisor.getGuestIds().isEmpty()) {
```

Comment:
Do we really need to check if it's empty?  Won't an empty list just bomb out of the loop immediately?